service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DiaryPlanStatementMapper.java

```diff
@@ -61,10 +62,28 @@
         buildEffectiveTime(procedureRequest).map(builder::effectiveTime);
         buildText(procedureRequest).map(builder::text);
         builder.code(buildCode(procedureRequest));
+        buildParticipant(procedureRequest, builder, idMapper);
 
         return Optional.of(TemplateUtils.fillTemplate(PLAN_STATEMENT_TEMPLATE, builder.build()));
     }
 
+    private void buildParticipant(
+            ProcedureRequest procedureRequest,
+            PlanStatementMapperParametersBuilder builder,
+            IdMapper idMapper
+    ) {
+        var requesterAgent = procedureRequest.getRequester().getAgent();
+
+        if (requesterAgent.hasReference()) {
+            var resourceType = requesterAgent.getReference().split("/")[0];
+            if (resourceType.equals(ResourceType.Practitioner.name())) {
+                String participantId = idMapper.getOrNew(procedureRequest.getRequester().getAgent());
```

Comment:
We've already got the agent in `requesterAgent`
```suggestion
                String participantId = idMapper.getOrNew(requesterAgent);
```

Reply:
done